integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsShutdownOption.java

```diff
@@ -270,6 +290,154 @@ void testShutdownPropsEnvOverride() {
+   * MS2: ignoreSessions -> true, waitForAllSessions -> true,
+   *      When ignoreSessions is set to true, waitForAllSessions does not apply
+   *
+   * Verify the shutdown options are set in the server log.
+   * Scale down the cluster and verify the operator will call REST API to shutdown the server before deleting the pod.
+   */
+  @Test
+  @DisplayName("Verify operator will call REST API to shutdown the server before deleting the pod")
+  void testShutdownServersUsingRestBeforeDelete() {
+
+    // create Shutdown objects for each server and cluster
+    Shutdown[] shutDownObjects = new Shutdown[5];
+    Shutdown dom = new Shutdown().ignoreSessions(Boolean.TRUE).shutdownType("Forced").timeoutSeconds(30L);
+    Shutdown admin = new Shutdown().ignoreSessions(Boolean.TRUE).shutdownType("Forced").timeoutSeconds(40L);
+    Shutdown cluster = new Shutdown().ignoreSessions(Boolean.FALSE).shutdownType("Graceful").timeoutSeconds(60L)
+        .waitForAllSessions(Boolean.TRUE);
+    Shutdown ms1 = new Shutdown().ignoreSessions(Boolean.FALSE).shutdownType("Graceful").timeoutSeconds(120L)
+        .waitForAllSessions(Boolean.TRUE);
+    Shutdown ms2 = new Shutdown().ignoreSessions(Boolean.TRUE).shutdownType("Graceful").timeoutSeconds(45L)
+        .waitForAllSessions(Boolean.TRUE);
+    shutDownObjects[0] = dom;
+    shutDownObjects[1] = admin;
+    shutDownObjects[2] = cluster;
+    shutDownObjects[3] = ms1;
+    shutDownObjects[4] = ms2;
+    // create domain custom resource and verify all the pods came up
+    Domain domain = buildDomainResource(shutDownObjects);
+    createVerifyDomain(domain);
+
+    // Verify values set above are present in the server log
+    verifyServerLog(domainNamespace, adminServerPodName,
+        new String[]{"SHUTDOWN_IGNORE_SESSIONS=true", "SHUTDOWN_TYPE=Forced", "SHUTDOWN_TIMEOUT=40"});
+    verifyServerLog(domainNamespace, managedServerPodNamePrefix + 1,
+        new String[]{"SHUTDOWN_IGNORE_SESSIONS=false", "SHUTDOWN_TYPE=Graceful", "SHUTDOWN_TIMEOUT=60",
+            "SHUTDOWN_WAIT_FOR_ALL_SESSIONS=true"});
+    verifyServerLog(domainNamespace, managedServerPodNamePrefix + 2,
+        new String[]{"SHUTDOWN_IGNORE_SESSIONS=false", "SHUTDOWN_TYPE=Graceful", "SHUTDOWN_TIMEOUT=60",
+            "SHUTDOWN_WAIT_FOR_ALL_SESSIONS=true"});
+    verifyServerLog(domainNamespace, indManagedServerPodName1,
+        new String[]{"SHUTDOWN_IGNORE_SESSIONS=false", "SHUTDOWN_TYPE=Graceful", "SHUTDOWN_TIMEOUT=120",
+            "SHUTDOWN_WAIT_FOR_ALL_SESSIONS=true"});
+    verifyServerLog(domainNamespace, indManagedServerPodName2,
+        new String[]{"SHUTDOWN_IGNORE_SESSIONS=true", "SHUTDOWN_TYPE=Graceful", "SHUTDOWN_TIMEOUT=45",
+            "SHUTDOWN_WAIT_FOR_ALL_SESSIONS=true"});
+
+    // scale down the cluster to 1 server
+    int newReplicaCount = replicaCount - 1;
+    logger.info("Scaling down the cluster {0} in namespace {1} to set the replicas to {2}",
```

Comment:
We have patching utility method. Can we use it, instead of explicit patch command.